legacy/build-deploy-docker-compose.sh

```diff
@@ -222,9 +233,61 @@ You can run docker compose config locally to check that your docker-compose file
 ##############################################"
   exit 1
 fi
-set -ex
 
-set +ex
+## validate the docker-compose in a way to eventually phase out forked library by displaying warnings
+dccOutput=$(bash -c 'build-deploy-tool validate docker-compose --ignore-non-string-key-errors=false --ignore-missing-env-files=false --docker-compose '${DOCKER_COMPOSE_YAML}'; exit $?' 2>&1)
+dccExit=$?
+if [ "${dccExit}" != "0" ]; then
+  ((++BUILD_WARNING_COUNT))
+  ((++DOCKER_COMPOSE_WARNING_COUNT))
+  echo "
+##############################################
+Warning!
+There are issues with your docker compose file that lagoon uses that should be fixed.
+You can run docker compose config locally to check that your docker-compose file is valid.
+"
+  if [[ "${dccOutput}" =~ "no such file or directory" ]]; then
+    echo "> an env_file is defined in your docker-compose file, but no matching file found."
+  fi
+  if [[ "${dccOutput}" =~ "Non-string key" ]]; then
+    echo "> an invalid string key was detected in your docker-compose file."
+  fi
+  echo ERR: ${dccOutput}
+  echo ""
+fi
+
+dccOutput=$(bash -c 'build-deploy-tool validate docker-compose-with-errors --docker-compose '${DOCKER_COMPOSE_YAML}'; exit $?' 2>&1)
+dccExit2=$?
+if [ "${dccExit2}" != "0" ]; then
+  ((++DOCKER_COMPOSE_WARNING_COUNT))
+  if [ "${dccExit}" == "0" ]; then
+    ((++BUILD_WARNING_COUNT))
+    echo "
+##############################################
+Warning!
+There are issues with your docker compose file that lagoon uses that should be fixed.
+You can run docker compose config locally to check that your docker-compose file is valid.
+"
+  fi
+  echo "> There are yaml validation errors in your docker-compose file that should be corrected."
+  echo ERR: ${dccOutput}
+  echo ""
+fi
+
+if [[ "$DOCKER_COMPOSE_WARNING_COUNT" -gt 0 ]]; then
+  echo "Read the docs for more on errors displayed here :LINK_TO_DOCS:
```

Comment:
I'll create two short links in https://github.com/uselagoon/lagoon/pull/3608
`https://docs.lagoon.sh/lagoon-build-errors` and
`https://docs.lagoon.sh/docker-compose-errors`
to point to a new page to house these errors